dandi/upload.py

```diff
@@ -573,3 +587,272 @@ def upload_agg(*ignored):
+            try:
+                metadata = get_metadata(path)
+            except Exception as exc:
+                if allow_any_path:
+                    yield {"status": "failed to extract metadata"}
+                    metadata = {}
+                else:
+                    yield skip_file("failed to extract metadata: %s" % str(exc))
+                    return
+
+            #
+            # Compute checksums and possible other digests (e.g. for s3, ipfs - TODO)
+            #
+            yield {"status": "digesting"}
+            try:
+                # TODO: in theory we could also cache the result, but since it is
+                # critical to get correct checksums, safer to just do it all the time.
+                # Should typically be faster than upload itself ;-)
+                digester = Digester(metadata_digests)
+                file_metadata_.update(digester(path))
+            except Exception as exc:
+                yield skip_file("failed to compute digests: %s" % str(exc))
+                return
+
+            #
+            # Determine metadata
+            #
+            metadata_ = {}
+            for k, v in metadata.items():
+                if v in ("", None):
+                    continue  # degenerate, why bother
+                # XXX TODO: remove this -- it is only temporary, search should handle
+                if isinstance(v, str):
+                    metadata_[k] = v.lower()
+                elif isinstance(v, datetime):
+                    metadata_[k] = ensure_strtime(v)
+            # we will add some fields which would help us with deciding to
+            # reupload or not
+            # .isoformat() would give is8601 representation but I see in girder
+            # already
+            # session_start_time   1971-01-01 12:00:00+00:00
+            # decided to go for .isoformat for internal consistency -- let's see
+            file_metadata_["uploaded_datetime"] = ensure_strtime(time.time())
+            metadata_.update(file_metadata_)
+            metadata_["uploaded_size"] = path_stat.st_size
+            metadata_["uploaded_mtime"] = ensure_strtime(path_stat.st_mtime)
+            metadata_["uploaded_by"] = "dandi %s" % __version__
+            # Also store object_id for the file to help identify changes/moves
+            try:
+                metadata_["uploaded_nwb_object_id"] = get_object_id(str(path))
+            except Exception as exc:
+                (lgr.debug if allow_any_path else lgr.warning)(
+                    "Failed to read object_id: %s", exc
+                )
+
+            #
+            # Upload file
+            #
+            yield {"status": "uploading"}
+            for r in client.iter_upload(dandiset, "draft", relpath, metadata_, path):
+                upload_perc = 100 * ((r["current"] / r["total"]) if r["total"] else 1.0)
+                uploaded_paths[str(path)]["size"] = r["current"]
+                yield {"upload": upload_perc}
```

Comment:
When exactly would the status be "producing asset"?

> "uploading" (in that case -- we are to yield this additional "upload" record for UI outside),

Do you mean that `"current"` and `"total"` should be placed inside an `"upload"` sub-dict when the status is "uploading"?

Reply:
> When exactly would the status be "producing asset"?

the `post` call to generate a new asset, as is done here https://github.com/dandi/dandi-cli/blob/master/dandi/dandiapi.py#L383 ATM

Reply:
> > "uploading" (in that case -- we are to yield this additional "upload" record for UI outside),
> 
> Do you mean that `"current"` and `"total"` should be placed inside an `"upload"` sub-dict when the status is "uploading"?

I just meant that instead of `yield {"upload": upload_perc}` to have `yield {"status": "uploading", "upload": upload_perc}`